model/riscv_insts_dext.sail

```diff
@@ -934,18 +934,14 @@ function clause execute (F_UN_TYPE_D(rs1, rd, FCLASS_D)) = {
 }
 
 function clause execute (F_UN_TYPE_D(rs1, rd, FMV_X_D)) = {
-  assert(xlen >= 64);
-  let rs1_val_D            = F(rs1)[63..0];
-  let rd_val_X  : xlenbits = sign_extend(rs1_val_D);
-  X(rd) = rd_val_X;
```

Comment:
Please keep the extends/truncates, they generalise the code to support xlen != 64 and flen != 64 (RV128 or RVQ), whereas yours assumes xlen == flen == 64.

Reply:
That is, the only code changes needed should be to add ` && flen >= 64` to the asserts.

Reply:
Agreed. This will need to be reverted in #445, so no reason to remove it here. Simplifying it to all happen in-line like you did with F seems cleaner though.

Reply:
Hmm good point! I think it was technically missing the `nan_box`, presumably because the Q version wasn't available here so I added it, and pulled the code from #448 to make the nan boxing functions support all of the float sizes.